machineconfiguration/v1alpha1/types_machineconfignode.go

```diff
@@ -0,0 +1,151 @@
+
+	// pool contains a reference to the MachineConfigPool that this MachineConfigNode's referenced Node belongs to.
+	// +kubebuilder:validation:Required
+	Pool MCOObjectReference `json:"pool"`
+
+	// configVersion holds the desired config version for the node targeted by this MachineConfigNode resource.
+	// The desired version represents the machine config the node will attempt to update to when it next has the chance.
+	// +kubebuilder:validation:Required
+	ConfigVersion MCNSpecMachineConfigVersion `json:"configVersion"`
+}
+
+// MachineConfigNodeStatus holds the reported information on a particular MachineConfigNode
+type MachineConfigNodeStatus struct {
+	// conditions represent the observations of a MachineConfigNode's current state.
+	// Known .status.conditions.type are: "UpdatePreparing", "UpdateInProgress", "UpdatePostAction", "UpdateCompleting", "Resuming", "Updated",
+	// "Ready", "ComparingMCs", "DrainingNode", "ApplyingFilesAndOS", "CordoningNode", "RebootingNode", and "ReloadingCRIO"
+	// +patchMergeKey=type
+	// +patchStrategy=merge
+	// +listType=map
+	// +listMapKey=type
+	Conditions []metav1.Condition `json:"conditions,omitempty" patchStrategy:"merge" patchMergeKey:"type"`
+	// observedGeneration represents the generation observed by the controller.
+	// +kubebuilder:validation:Required
+	ObservedGeneration int64 `json:"observedGeneration,omitempty"`
+	// configVersion describes the current and desired MachineConfig for this node.
+	// The current version represents the current machine config for the node and is updated after a successful update.
+	// The desired version represents the machine config the node will attempt to update to when it next has the chance.
+	// +kubebuilder:validation:Required
+	ConfigVersion MCNStatusMachineConfigVersion `json:"configVersion"`
+}
+
+// MCNStatusMachineConfigVersion holds the current and desired config versions as last updated in the MCN status.
+// When Current != Desired; the MachineConfigOperator is in an upgrade phase and the MachineConfigNode will
+// take account of upgrade related events. Otherwise they will be ignored given that certain operations
+// happen both during the MCO's upgrade mode and the daily operations mode.
+type MCNStatusMachineConfigVersion struct {
+	// current is the current MachineConfig a node is using
+	// The machine state controller sets this value once the MCO has completed an update to the configuration of the node.
+	// Must be a lowercase RFC-1123 hostname (https://tools.ietf.org/html/rfc1123)
+	// +kubebuilder:validation:MaxLength=253
+	// +kubebuilder:validation:Pattern=`^([a-zA-Z0-9]|[a-zA-Z0-9][a-zA-Z0-9\-]{0,61}[a-zA-Z0-9])(\.([a-zA-Z0-9]|[a-zA-Z0-9][a-zA-Z0-9\-]{0,61}[a-zA-Z0-9]))*$`
+	// +optional
+	Current string `json:"current"`
+	// desired is the MachineConfig the node wants to upgrade to.
+	// The machine state controller sets this value once the MCO has generated a new MachineConfig for a node to upgrade to.
```

Comment:
The status version of this field, should talk about being updated once the MCD has accepted that it can move to this new version right?